docs/contracts/staking-router.md

```diff
@@ -3,9 +3,843 @@
+    
+4. If there are staking modules available (`stakingModulesCount > 0`), the function goes into the allocation process:
+    
+    a. It calculates a new estimated total of active validators, adding the new deposits to the total active validators (`totalActiveValidators += _depositsToAllocate`).
+    
+    b. It creates a `capacities` array of the same size as the number of staking modules. Each entry in this array represents the maximum capacity of a particular staking module, i.e., the maximum number of validators that module can have. This is calculated as the minimum of either:
+    
+    - The target number of validators for a module, which is based on a desired target share (`stakingModulesCache[i].targetShare * totalActiveValidators / TOTAL_BASIS_POINTS`), or
+    - The sum of the current active validators and the available validators in the module (`stakingModulesCache[i].activeValidatorsCount + stakingModulesCache[i].availableValidatorsCount`).
+    
+    c. Finally, it calls the `allocate` function from `MinFirstAllocationStrategy`, passing in the `allocations`, `capacities`, and `_depositsToAllocate`. The amount successfully allocated is stored in `allocated`.
+    
+
+To sum up, this function is using the `MinFirstAllocationStrategy` algorithm to distribute new deposits (validators) across different staking modules in a way that prioritizes filling the least populated modules, while taking into account each module's target share and capacity. The resulting allocations and total allocated amount are then returned for further use.
+
+## Fee distribution
+
+The fee structure is set independently in each module. There are two components to the fee structure: the module fee and the treasury fee, both specified as percentages (basis points). For example, a 5% (500 basis points) module fee split between node operators in the module and a 5% (500 basis points) treasury fee sent to the treasury. Additionally, `StakingRouter` utilizes a precision factor of 100 * 1018 for fees that prevents arithmetic operations from truncating the fees of small modules.
+
+Because Lido does not account for validator performance, the protocol fee is distributed between modules proportionally to active validators and the specified module fee. For example, a module with 75% of all validators in the protocol and a 5% module fee will receive 3.75% of the total rewards across the protocol. This means that if the modules' fee and treasury fee do not exceed 10%, the total protocol fee will not either, no matter how many modules there are. There is also an edge case where the module is stopped for emergency while its validators are still active. In this case the module fee will be transferred to the treasury and once the module is back online, the rewards will be returned back to the module from the treasury.
```

Comment:
`Because Lido does not account for validator performance`
to
`Because protocol currently does not account for per-validator performance`